app/src/modules/settings/routes/data-model/fields/fields.vue

```diff
@@ -24,15 +23,16 @@ const { t } = useI18n();
 const router = useRouter();
 
 const { collection } = toRefs(props);
-const { info: collectionInfo } = useCollection(collection);
 const collectionsStore = useCollectionsStore();
 const fieldsStore = useFieldsStore();
 
-const { edits, item, saving, loading, save, remove, deleting } = useItem(ref('directus_collections'), collection);
+// Re-initialize if collection (page) changes
+const usableItem = computed(() => useItem(ref('directus_collections'), collection));
```

Comment:
I dont think wrapping the already reactive return values of `useItem` in an extra computed is a better approach than the previous one. Why was this entire PR rewritten after a small piece of feedback with an easy fix? 🤔

Reply:
Should we not fix this at the source then in `useItems` so it properly resets its internal state when the input refs `collection`, `primaryKey` or `query` change? Putting `useItem` in a computed watching the same variables as it does internally feels like treating the symptoms too 😬

Reply:
Fair point! I hesitated to directly adjust `useItem` because it is used in several places and in different ways, but I took a closer look now and found no problem with that.

Reply:
Very understandable, I was indeed afraid that changing `useItem` would be regression/bug prone but it did sound like the right place to fix it so glad you took a closer look 😄